compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -435,7 +435,10 @@ class Namer { typer: Typer =>
     *  package members are not entered twice in the same run.
     */
   def enterSymbol(sym: Symbol)(using Context): Symbol = {
-    if (sym.exists) {
+    // We do not enter Scala 2 macros defined in Scala 3.
+    // Except if they are defined in StringContext to allow the compilation of the stdlib
+    //   dotty.communitybuild.CommunityBuildTest.stdLib213
+    if (sym.exists && (!sym.isScala2MacroInScala3 || sym.owner == defn.StringContextClass)) {
```

Comment:
Later, if we decide to keep this scheme we should add an inline variant of `StringContext.s`, `StringContext.raw` and `StringContext.f`. The contidtion would be
```scala
if (sym.exists && !sym.isScala2MacroInScala3)
```

Reply:
I'd add that note to the comment in the code.